From bdd2f2b8c3fbb68fd935b544d8dddacb3fea1066 Mon Sep 17 00:00:00 2001
From: two <two@envs.net>
Date: Sun, 11 Sep 2022 12:47:32 +0000
Subject: [PATCH] mkdir -p webkit-web-extension

create webextension dir to prevent webprocess crash when it's not found
---
 badwolf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/badwolf.c b/badwolf.c
index 2fdb18e..4c0bdec 100644
--- a/badwolf.c
+++ b/badwolf.c
@@ -670,6 +670,7 @@ new_browser(struct Window *window, const gchar *target_url, struct Client *old_b
 		webkit_web_context_set_process_model(web_context,
 		                                     WEBKIT_PROCESS_MODEL_MULTIPLE_SECONDARY_PROCESSES);
 		webkit_web_context_set_web_extensions_directory(web_context, web_extensions_directory);
+		g_mkdir_with_parents(web_extensions_directory, 0777);

 		g_signal_connect(G_OBJECT(web_context),
 		                 "download-started",
--
2.37.2

